napari/_qt/qt_viewer.py

```diff
@@ -754,9 +755,19 @@ def _map_canvas2world(self, position):
             Position in world coordinates, matches the total dimensionality
             of the viewer.
         """
+        # nd = self.viewer.dims.ndisplay
+        #
+        # transform = self.view.camera.transform.inverse
+        # mapped_position = transform.map(list(position))[:nd]
+        # position_world_slice = mapped_position[::-1]
+        #
+        # position_world = list(self.viewer.dims.point)
+        # for i, d in enumerate(self.viewer.dims.displayed):
+        #     position_world[d] = position_world_slice[i]
+
         nd = self.viewer.dims.ndisplay
-        transform = self.view.camera.transform.inverse
-        mapped_position = transform.map(list(position))[:nd]
+        transform = self.view.scene.transform
```

Comment:
@alisterburt , what do you think about this method? I think it's much more intuitive and easier to read than the previous method. I'm not sure about performance though...

Reply:
yeah nice, this looks great, much easier to read! I'm not worried about performance given that the number of displayed dims will be small